pints/tests/test_mcmc_differential_evolution.py

```diff
@@ -173,9 +173,8 @@ def test_set_hyper_parameters(self):
             ValueError, 'non-negative', mcmc.set_hyper_parameters,
             [1, -0.5, 20, 0, 0])
 
-        self.assertRaisesRegex(
-            ValueError, 'integer', mcmc.set_hyper_parameters,
-            [1, 0.5, 20.5, 0, 0])
+        mcmc.set_hyper_parameters([0.5, 0.6, 20.2, 0, 0])
+        self.assertEqual(mcmc._gamma_switch_rate, 20)
```

Comment:
This should still be checked for the direct method.
In fact, don't really see why we're checking these methods via `set_hyper_parameters` instead of directly!